modules/oe_editorial_corporate_workflow/modules/oe_editorial_corporate_workflow_translation/tests/src/Functional/CorporateWorkflowTranslationRevisionTest.php

```diff
@@ -549,4 +552,95 @@ public function testTranslationRevisionsCarryOver(): void {
+   */
+  public function testModerationStateSync(): void {
+    // Create a validated node and add a translation.
+    /** @var \Drupal\node\NodeStorageInterface $node_storage */
+    $node_storage = $this->entityTypeManager->getStorage('node');
+
+    /** @var \Drupal\node\NodeInterface $node */
+    $node = $node_storage->create([
+      'type' => 'page',
+      'title' => 'My node',
+      'moderation_state' => 'draft',
+    ]);
+    $node->save();
+    $node = $this->moderateNode($node, 'validated');
+    $this->drupalGet(Url::fromRoute('oe_translation.permission_translator.create_local_task', [
+      'entity' => $node->id(),
+      'source' => 'en',
+      'target' => 'fr',
+      'entity_type' => 'node',
+    ]));
+    $this->assertSession()->elementContains('css', '#edit-title0value-translation', 'My node');
+    $values = [
+      'Translation' => 'My node FR',
+    ];
+    // It should be the first local task item created so we use the ID 1.
+    $url = Url::fromRoute('entity.tmgmt_local_task_item.canonical', ['tmgmt_local_task_item' => 1]);
+    $this->drupalGet($url);
+    $this->submitForm($values, t('Save and complete translation'));
+
+    // Assert that the node has two translations and the moderation state entity
+    // also has two translations.
+    $node_storage->resetCache();
+    /** @var \Drupal\node\NodeInterface $node */
+    $node = $node_storage->load($node->id());
+    $this->assertCount(2, $node->getTranslationLanguages());
+    $moderation_state = ContentModerationState::loadFromModeratedEntity($node);
+    $this->assertCount(2, $moderation_state->getTranslationLanguages());
+
+    // Assert that both the node and moderation state translations are
+    // "validated".
+    $assert_translation_state = function (ContentEntityInterface $entity, $state) {
+      foreach ($entity->getTranslationLanguages() as $language) {
+        $translation = $entity->getTranslation($language->getId());
+        $this->assertEquals($state, $translation->get('moderation_state')->value, sprintf('The %s language has the %s state', $language->getName(), $state));
+      }
+    };
+    $assert_translation_state($node, 'validated');
+    $assert_translation_state($moderation_state, 'validated');
+
+    // "Break" the system by deleting the moderation state entity translation.
+    $moderation_state->removeTranslation('fr');
+    ContentModerationStateEntity::updateOrCreateFromEntity($moderation_state);
+
+    // Now only the original will be validated, and the translation of the node
+    // becomes "draft" because it no longer is translated. This situation
+    // should not really occur, but if it does, it can break new translations
+    // which when being saved onto the node, cause the moderation state of
+    // the original to be set to draft instead of keeping it on validated.
+    $node_storage->resetCache();
+    /** @var \Drupal\node\NodeInterface $node */
```

Comment:
This can be removed.

Reply:
No it is not...entity storage returns EntityInterface, so I just type hinted.